### PR TITLE
Add support for Default Baggage Injection in URLSessionInstrumentation

### PR DESCRIPTION
**Pull request by batuhansk**
**Description:**
This PR introduces a `defaultBaggageProvider` callback in `URLSessionInstrumentationConfiguration`, enabling a default baggage to be applied globally to all instrumented requests. The provided baggage is combined with active baggage when available, or used standalone otherwise. The `tracePropagationHTTPHeaders` method has been optimized to ensure clear and concise baggage propagation across all requests.

### CODE REVIEW

File: Tests/InstrumentationTests/URLSessionTests/URLSessionInstrumentationTests.swift [hunk 1]
```diff
@@ -250,7 +260,7 @@ class URLSessionInstrumentationTests: XCTestCase {
         XCTAssertTrue(URLSessionInstrumentationTests.checker.receivedErrorCalled)
     }
 
-    public func testShouldInstrumentRequest() throws {
+    public func testShouldInstrumentRequest_PropagateCombinedActiveAndDefaultBaggages() throws {
```
Comment (nachoBonafonte) [inline Tests/InstrumentationTests/URLSessionTests/URLSessionInstrumentationTests.swift:263]: Could this functionally be tested in different tests? Just move baggage evaluation to two new different tests, and keep the existing test without baggage evaluation

**Review by nachoBonafonte — CHANGES_REQUESTED** [review]
LGTM, just a small change would be desired

File: Tests/InstrumentationTests/URLSessionTests/URLSessionInstrumentationTests.swift (hunk 1 above)
Comment (batuhansk) [reply Tests/InstrumentationTests/URLSessionTests/URLSessionInstrumentationTests.swift:281]: Makes sense! Baggage logic is now in separate tests, and the original test stays focused on the basics.

**Review by nachoBonafonte — APPROVED** [review]
Thanks for updating, looks perfect for me now.

**Review by vvydier — APPROVED** [review]

**Pull request merged**